### PR TITLE
Feature/16 body parsing

### PR DESCRIPTION
**Pull request by shoko-soeno**
This pull request introduces robust support for HTTP request body parsing, specifically for both `Content-Length` and `chunked` transfer encoding modes. It refactors the `HttpRequest` class to better handle streaming and partial data, adds comprehensive helper methods for chunked parsing, and includes extensive tests to cover normal, edge, and error scenarios. Additionally, a minor build system fix is included for Linux platforms.

**HTTP Request Body Parsing Improvements**

* Added new methods in `HttpRequest` for parsing body data: `AdvanceBodyParsing`, `AdvanceContentLengthBody`, `AdvanceChunkedBody`, and chunked parsing helpers (`ParseChunkSize`, `ValidateFinalCRLF`, `AppendChunkData`). These methods allow incremental, robust parsing of both content-length and chunked bodies, handling partial data and errors gracefully. [[1]](diffhunk://#diff-125dde9589a6b8dc46a83592e0ff80f7da375aac59a64ff92cbd87b5d321e788R62-R67) [[2]](diffhunk://#diff-548ef5f334f94d690920a36413351686bfc1c8a38bd1175aa6f258cbe45355aeR1-R146)
* Refactored the `HttpRequest` class fields and constructors to support chunked parsing state, including `chunked_parsed_bytes_` and `pending_chunk_bytes_`, and moved the `Progress` enum and parsing state fields to public for better testability and clarity. [[1]](diffhunk://#diff-125dde9589a6b8dc46a83592e0ff80f7da375aac59a64ff92cbd87b5d321e788L33-L51) [[2]](diffhunk://#diff-125dde9589a6b8dc46a83592e0ff80f7da375aac59a64ff92cbd87b5d321e788R82-R98) [[3]](diffhunk://#diff-b3964ffacd70fff3f96326d6ec6c93b8d80035be8130c5c677705f5fd363d534L10-R23)

**Testing**

* Added a comprehensive new test file `tests/HttpRequest_advanceBodyParsing.test.cpp` covering happy paths, partial/streaming scenarios, malformed requests, edge cases, and boundary conditions for both content-length and chunked body parsing.

**Build System**

* Updated `CMakeLists.txt` to set the `-no-pie` linker flag for Linux builds, ensuring compatibility and consistent executable generation.

### CODE REVIEW

File: srcs/parse/HttpRequest_advanceBodyParsing.cpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+          progress_ = kDone;
+        }
+        return done;
+      }
+    }
+    size_t pos = chunked_parsed_bytes_;
+    if (!AppendChunkData(pos, static_cast<size_t>(pending_chunk_bytes_))) {
+      return false;
+    }
+    chunked_parsed_bytes_ = pos;
```
[the remark this answers is not in the document]
Comment (shoko-soeno) [reply srcs/parse/HttpRequest_advanceBodyParsing.cpp]: I think this feedback is already outdated. 
I have checked this bug fixed in test such as 
AdvanceBodyParsing_Chunked_PartialRecv_DoesNotDuplicateBody

**Review by yuzu-juice — APPROVED** [review]
修正ありがとうございました。

以下の二つのどちらかがいいと思いました。
- 最初の PR の、public に test 用の method を追加する方針でやる
- friend keyword を使うと、もっと簡潔に書けそう (webserv だと friend 使って良さそう)

ちょっと他にいいやり方を思いつかなかったです。

どうするかは、お任せします
